pkg/signature/algorithm_registry_test.go

```diff
@@ -0,0 +1,170 @@
+//
+// Copyright 2024 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package signature
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/rsa"
+	"testing"
+
+	v1 "github.com/sigstore/protobuf-specs/gen/pb-go/common/v1"
+)
+
+func TestGetAlgorithmDetails(t *testing.T) {
+	details, err := GetAlgorithmDetails(v1.KnownSignatureAlgorithm_ECDSA_SHA2_256_NISTP256)
+	if err != nil {
+		t.Errorf("unexpected error getting algorithm details: %v", err)
+	}
+	if details.GetSignatureAlgorithm() != v1.KnownSignatureAlgorithm_ECDSA_SHA2_256_NISTP256 {
+		t.Errorf("unexpected signature algorithm")
+	}
+	if details.GetKeyType() != ECDSA {
+		t.Errorf("unexpected key algorithm")
+	}
+	if details.GetHashType() != crypto.SHA256 {
+		t.Errorf("unexpected hash algorithm")
+	}
+	curve, err := details.GetECDSACurve()
+	if err != nil {
+		t.Errorf("unexpected error getting ecdsa curve")
+	}
+	if (*curve) != elliptic.P256() {
+		t.Errorf("unexpected curve")
+	}
+	_, err = details.GetRSAKeySize()
+	if err == nil {
+		t.Errorf("unexpected success getting rsa key size")
+	}
+}
+
+func TestAlgorithmRegistryConfig(t *testing.T) {
+	config, err := NewAlgorithmRegistryConfig([]v1.KnownSignatureAlgorithm{v1.KnownSignatureAlgorithm_ECDSA_SHA2_256_NISTP256, v1.KnownSignatureAlgorithm_ED25519})
+	if err != nil {
+		t.Errorf("unexpected error creating algorithm registry config: %v", err)
+	}
+
```

Comment:
Can we also add a test for RSA keys being permitted?

It might be overkill, but ideally we'd have a test for every supported key type, to have maximum code coverage.

Reply:
[c7acffd](https://github.com/sigstore/sigstore/pull/1601/commits/c7acffd09c81246401a30270fe6605d18c5713ea)